core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/DataSynchronizer.java

```diff
@@ -91,6 +97,75 @@
  * documents.
  */
 public class DataSynchronizer implements NetworkMonitor.StateListener {
+  class BatchOps {
```

Comment:
[nit] May be better to move this into its own class in a different file instead of this being an inner non-static class. Would just need to provide more args for its methods though to compensate not having direct access. Might be cleaner though.